### ISSUE TITLE
Hi, issue during 'make' command about minisam::Variables& minisam::Variables::operator=(const minisam::Variables&)

### ISSUE THREAD

**Issue by lfwin** [dongjing3309/minisam#33]
Hi,

During compiling the code, the following errors occurs, 
_
[  2%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/DoglegOptimizer.cpp.o
[  4%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/GaussNewtonOptimizer.cpp.o
[  6%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/LevenbergMarquardtOptimizer.cpp.o
[  8%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/NonlinearOptimizer.cpp.o
[ 11%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/NumericalFactor.cpp.o
[ 13%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/SparsityPattern.cpp.o
[ 15%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/linearization.cpp.o
[ 17%] Building CXX object minisam/CMakeFiles/minisam.dir/utils/Timer.cpp.o
[ 20%] Building CXX object minisam/CMakeFiles/minisam.dir/utils/testAssertions.cpp.o
/home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp: In member function ‘minisam::NonlinearOptimizationStatus minisam::DoglegOptimizer::tryRadius_(const minisam::FactorGraph&, minisam::Variables&, const VectorXd&, const VectorXd&, double, double)’:
/home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp:140:75: error: implicitly-declared ‘minisam::Variables& minisam::Variables::operator=(const minisam::Variables&)’ is deprecated [-Werror=deprecated-copy]
  140 |       values_update = values.retract(dx_dl, h_sparsity_cache_.var_ordering);
      |                                                                           ^
In file included from /home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp:11:
/home/win/work/minisam/minisam/core/Variables.h:33:3: note: because ‘minisam::Variables’ has user-provided ‘minisam::Variables::Variables(const minisam::Variables&)’
   33 |   Variables(const Variables& variables);
      |   ^~~~~~~~~
/home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp:142:75: error: implicitly-declared ‘minisam::Variables& minisam::Variables::operator=(const minisam::Variables&)’ is deprecated [-Werror=deprecated-copy]
  142 |       values_update = values.retract(dx_dl, j_sparsity_cache_.var_ordering);
      |                                                                           ^

_

My system is Ubuntu 20.04 with cuda installed as version 11.5.

**Comment by ksy1113** [comment]
I have the same issue

**Comment by ksy1113** [comment]
> Hi,
> 
> During compiling the code, the following errors occurs, _ [ 2%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/DoglegOptimizer.cpp.o [ 4%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/GaussNewtonOptimizer.cpp.o [ 6%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/LevenbergMarquardtOptimizer.cpp.o [ 8%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/NonlinearOptimizer.cpp.o [ 11%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/NumericalFactor.cpp.o [ 13%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/SparsityPattern.cpp.o [ 15%] Building CXX object minisam/CMakeFiles/minisam.dir/nonlinear/linearization.cpp.o [ 17%] Building CXX object minisam/CMakeFiles/minisam.dir/utils/Timer.cpp.o [ 20%] Building CXX object minisam/CMakeFiles/minisam.dir/utils/testAssertions.cpp.o /home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp: In member function ‘minisam::NonlinearOptimizationStatus minisam::DoglegOptimizer::tryRadius_(const minisam::FactorGraph&, minisam::Variables&, const VectorXd&, const VectorXd&, double, double)’: /home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp:140:75: error: implicitly-declared ‘minisam::Variables& minisam::Variables::operator=(const minisam::Variables&)’ is deprecated [-Werror=deprecated-copy] 140 | values_update = values.retract(dx_dl, h_sparsity_cache_.var_ordering); | ^ In file included from /home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp:11: /home/win/work/minisam/minisam/core/Variables.h:33:3: note: because ‘minisam::Variables’ has user-provided ‘minisam::Variables::Variables(const minisam::Variables&)’ 33 | Variables(const Variables& variables); | ^~~~~~~~~ /home/win/work/minisam/minisam/nonlinear/DoglegOptimizer.cpp:142:75: error: implicitly-declared ‘minisam::Variables& minisam::Variables::operator=(const minisam::Variables&)’ is deprecated [-Werror=deprecated-copy] 142 | values_update = values.retract(dx_dl, j_sparsity_cache_.var_ordering); | ^
> 
> _
> 
> My system is Ubuntu 20.04 with cuda installed as version 11.5.

did you solve the issue?

**Comment by lfwin** [comment]
yes, I add a copy constructor in Variables.h behind default constructor as follows,
`
  Variables operator=( Variables variables) {
    return variables;
  } ;
`

**Comment by zizhengu** [comment]
> yes, I add a copy constructor in Variables.h behind default constructor as follows, `Variables operator=( Variables variables) { return variables; } ;`

Hi! I encountered this problem too.  I tried fix it in your way but I am not familiar with C++.  If it is possible, can I get your **Variables.h** file? Tanks very much!
My email: zizhengu@163.com
